tests/admin_app_tests/endpoint_tests/images_test.go

```diff
@@ -0,0 +1,222 @@
+}
+
+func TestPostImageNotAnImageFile(t *testing.T) {
+	database_mock := mocks.DatabaseMock{
+		AddImageHandler: func(id string, file_name string, alt_text string) error {
+			return nil
+		},
+	}
+
+	r := admin_app.SetupRoutes(app_settings, database_mock)
+	w := httptest.NewRecorder()
+
+	pr, pw := io.Pipe()
+	writer := multipart.NewWriter(pw)
+
+	go func() {
+		defer writer.Close()
+
+		part, err := createTestForm(writer, "file", "test.png", "image/png")
+		if err != nil {
+			t.Error(err)
+		}
+
+		text := bytes.NewBufferString("This is some dumy text to check the content test")
+		_, err = io.Copy(part, text)
+		if err != nil {
+			t.Error(err)
+		}
+	}()
+
+	req, _ := http.NewRequest("POST", "/images", pr)
+	req.Header.Add("Content-Type", writer.FormDataContentType())
+	r.ServeHTTP(w, req)
+
+	assert.Equal(t, 400, w.Code)
+}
+
+func TestPostImageWrongFileContentType(t *testing.T) {
+	database_mock := mocks.DatabaseMock{
+		AddImageHandler: func(id string, file_name string, alt_text string) error {
+			return nil
+		},
+	}
+
+	r := admin_app.SetupRoutes(app_settings, database_mock)
+	w := httptest.NewRecorder()
+
+	pr, pw := io.Pipe()
+	writer := multipart.NewWriter(pw)
+
+	go func() {
+		defer writer.Close()
+
+		part, err := createTestForm(writer, "file", "test.png", "application/json")
+		if err != nil {
+			t.Error(err)
+		}
+
+		img := createImage()
+
+		if err != nil {
+			t.Error(err)
+		}
+
+		err = png.Encode(part, img)
+		if err != nil {
+			t.Error(err)
+		}
+	}()
+
+	req, _ := http.NewRequest("POST", "/images", pr)
+	req.Header.Add("Content-Type", writer.FormDataContentType())
+	r.ServeHTTP(w, req)
+
+	assert.Equal(t, 400, w.Code)
+}
+
+func TestPostImageFailedToCreateDatabaseEntry(t *testing.T) {
+	database_mock := mocks.DatabaseMock{
+		AddImageHandler: func(id string, file_name string, alt_text string) error {
+			return errors.New("test error")
+		},
+	}
+
+	r := admin_app.SetupRoutes(app_settings, database_mock)
+	w := httptest.NewRecorder()
+
+	pr, pw := io.Pipe()
+	writer := multipart.NewWriter(pw)
+
+	go func() {
+		defer writer.Close()
+
+		part, err := createTestForm(writer, "file", "test.png", "image/png")
+		if err != nil {
+			t.Error(err)
+		}
+
+		img := createImage()
+		if err != nil {
+			t.Error(err)
+		}
+
+		err = png.Encode(part, img)
+		if err != nil {
+			t.Error(err)
+		}
+	}()
+
+	req, _ := http.NewRequest("POST", "/images", pr)
+	req.Header.Add("Content-Type", writer.FormDataContentType())
+	r.ServeHTTP(w, req)
+
+	assert.Equal(t, 500, w.Code)
+
+}
+
+func createTestForm(writer *multipart.Writer, fieldname string, filename string, contentType string) (io.Writer, error) {
+	h := make(textproto.MIMEHeader)
+	h.Set("Content-Disposition",
+		fmt.Sprintf(`form-data; name="%s"; filename="%s"`, fieldname, filename))
+	h.Set("Content-Type", contentType)
+	return writer.CreatePart(h)
+}
+
+// Creating an image in memory for testing: https://yourbasic.org/golang/create-image/
+func createImage() image.Image {
```

Comment:
May I suggest even simpler function: just create a 1x1 pixel image 😉   - That way you will only need like one line and also no for loop. We only care about the `png` headers added to the image for the tests really

Reply:
Not required, if you see benefits in having a bigger image, we go for it !

Reply:
I've removed the entire creation of the image and revert a blank image without any pixels. As you mentioned already, only the meta data are important for us.